components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/MonitoredItemMessageEncoder.cs

```diff
@@ -312,8 +316,17 @@ public class MonitoredItemMessageEncoder : IMessageEncoder {
         /// </summary>
         /// <param name="messages"></param>
         /// <param name="encoding"></param>
+        /// <param name="context"></param>
         private IEnumerable<MonitoredItemMessage> GetMonitoredItemMessages(
-            IEnumerable<DataSetMessageModel> messages, MessageEncoding encoding) {
+            IEnumerable<DataSetMessageModel> messages, MessageEncoding encoding,
+            ServiceMessageContext context) {
+            if (context?.NamespaceUris == null) {
+                // declare all notifications in messages dropped 
+                foreach (var message in messages) {
+                    NotificationsDroppedCount += (uint)message?.Notifications?.Count();
```

Comment:
If message or Notifications is null, it will result in null and that cannot be added to the count. Should be:
NotificationsDroppedCount += (uint)message?.Notifications?.Count() ?? 0;

Reply:
thanks!